test/src/test/integration/link_git_protocol.rs

```diff
@@ -6,91 +6,106 @@
-        repo.find_tree(oid).unwrap()
-    };
-    let base = {
-        let oid = repo
-            .commit(
-                Some("refs/namespaces/foo/refs/heads/main"),
-                &auth,
-                &auth,
-                "initial",
-                &tree,
-                &[],
-            )
-            .unwrap();
-        repo.find_commit(oid).unwrap()
+    let repo = git::init_bare(&tmp).unwrap().into_easy();
+    let auth = git::actor::Signature::now_local_or_utc("apollo", "apollo@cree.de");
+
+    let empty_tree_id = repo
+        .write_object(&git::objs::Tree::empty().into())
+        .unwrap()
+        .detach();
+    let base_id = {
+        repo.commit(
+            "refs/namespaces/foo/refs/heads/main",
+            "initial",
+            auth.clone(),
+            auth.clone(),
+            empty_tree_id,
+            git::commit::NO_PARENT_IDS,
+        )
+        .unwrap()
     };
-    let next = repo
+    let next_commit_id = repo
         .commit(
-            Some("refs/namespaces/foo/refs/heads/next"),
-            &auth,
-            &auth,
+            "refs/namespaces/foo/refs/heads/next",
             "ng",
-            &tree,
-            &[&base],
+            auth.clone(),
+            auth,
+            empty_tree_id,
+            [base_id],
         )
-        .unwrap();
+        .unwrap()
+        .detach();
     repo.reference(
         "refs/namespaces/foo/refs/pulls/1/head",
-        next,
-        true,
+        next_commit_id,
+        PreviousValue::Any,
         "pee arrr",
     )
     .unwrap();
 
     tmp
 }
 
-fn collect_refs(repo: &git2::Repository) -> Result<Vec<(String, git2::Oid)>, git2::Error> {
+#[allow(dead_code)]
+fn collect_refs(repo: &impl git::easy::Access) -> anyhow::Result<Vec<git::refs::Reference>> {
     repo.references()?
-        .map(|x| x.map(|r| (r.name().unwrap().to_owned(), r.target().unwrap())))
+        .all()?
+        .map(|x| x.map(|r| r.detach()).map_err(|err| anyhow::anyhow!(err)))
         .collect()
 }
 
-fn update_tips<'a, T>(repo: &git2::Repository, tips: T) -> Result<(), anyhow::Error>
+#[allow(dead_code)]
+fn update_tips<'a, T>(repo: &git::Easy, tips: T) -> Result<(), anyhow::Error>
 where
     T: IntoIterator<Item = &'a Ref>,
 {
-    for r in tips {
-        match r {
-            Ref::Direct { path, object } => {
-                repo.reference(
-                    path.to_str()?,
-                    git2::Oid::from_bytes(object.as_slice())?,
-                    true,
-                    "",
-                )?;
-            },
+    use std::convert::TryInto;
+    let edits = tips
+        .into_iter()
+        .map(|r| match r {
+            Ref::Direct { path, object } => Ok(RefEdit {
+                change: Change::Update {
+                    log: Default::default(),
+                    expected: PreviousValue::Any,
+                    new: git::hash::ObjectId::from(object.as_slice()).into(), // TODO: remove type conversion when possible
```

Comment:
Tried again to remove this type conversion, to no avail. I don't understand why `git_hash::ObjectId` are not the same. Sure, they come from different crates but they are from the same source, `git_hash` in the same version.

Reply:
> git_hash in the same version.

That is a bit concerning -- I'm not sure I understand how Byron/gitoxide#192 would explain this.

Reply:
🤦‍♂️ fixed!
There was a `git-hash` v0.5 lingering around and I didn't notice it when checking `cargo tree -v git-hash`. Now they are all the same version and `cargo` behaves as `cargo` behaves.